src/components/elements/CommonHtmlContent.tsx

```diff
@@ -1,14 +1,20 @@
 import { Typography, TypographyProps } from '@mui/material';
+import DOMPurify from 'dompurify';
+import { useMemo } from 'react';
 
 interface Props extends TypographyProps {
   children: string;
 }
 const CommonHtmlContent: React.FC<Props> = ({ children, ref, ...props }) => {
+  const purifiedHtml = useMemo(() => {
+    return DOMPurify.sanitize(children);
+  }, [children]);
+
   return children ? (
     <Typography
-      {...props}
       component='div'
-      dangerouslySetInnerHTML={{ __html: children }}
+      {...props}
```

Comment:
I moved `props` under `component` here because the `RequiredLabel` passes in `component='span'`. I think it makes sense to keep `div` as default but allow it to be overridden.

Reply:
I made `component` an explicit prop to try to solve this build failure on the `tsc` step:
```
Error: src/modules/form/components/RequiredLabel.tsx(17,63): error TS2322: Type '{ children: string; component: string; border?: ResponsiveStyleValue<number | "medium" | "none" | (string & {}) | "inset" | "hidden" | "-moz-initial" | "inherit" | "initial" | "revert" | "revert-layer" | ... 188 more ... | undefined> | ((theme: Theme) => ResponsiveStyleValue<...>); ... 363 more ...; onTransitionEndC...' is not assignable to type 'IntrinsicAttributes & Props'.
  Property 'component' does not exist on type 'IntrinsicAttributes & Props'.
```

I was surprised about this though, because the `CommonHtmlContent`'s `Props` extends `TypographyProps`, so I would expect `component` to be allowed, and adding it explicitly is a bit redundant. Let me know if you think I've taken an OK approach here or we should change things around.

(Also, I noticed that running `yarn tsc` locally with the failing code resulted in not this error but a couple of other errors in `apolloClient`, `ApolloErrorAlert`, and `GenericTableWithData` - is this true for others too? Anything to worry about?)

Reply:
> I was surprised about this though, because the CommonHtmlContent's Props extends TypographyProps, so I would expect component to be allowed, and adding it explicitly is a bit redundant. Let me know if you think I've taken an OK approach here or we should change things around.

Thanks for calling this out, that is odd, it doesn't seem like declaring the component prop should necessary. 

There might be something funny going on with our typescript dependency; I see we have several versions of typescript installed. Seems like we should do some investigation,[ added issue for it here](https://www.pivotaltracker.com/story/show/187142123)

Reply:
Strange, thanks for flagging!